### PR TITLE
[POA] Fix rewinding expired polls

### PR DESCRIPTION
**Pull request by fassadlr**

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.PoA/Voting/PollsRepository.cs [hunk 1]
```diff
@@ -153,6 +151,16 @@ public void Initialize()
                                 modified = true;
                             }
 
+                            // Check if the current tip is before the poll expiry activation block,
+                            // if so un-expire it.
+                            if (poll.IsExpired && this.CurrentTip.Height < this.network.ConsensusOptions.Release1100ActivationHeight)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.PoA/Voting/PollsRepository.cs:156]: @quantumagi this part is important to review.

Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.PoA/Voting/PollsRepository.cs:156]:
```
if (poll.IsExpired && !PollIsExpiredAt(poll, this.CurrentTip, this.network))
{
  poll.IsExpired = false;
  ...
}

public static bool IsPollExpiredAt(Poll poll, ChainedHeader chainedHeader, Network network)
{
    if (chainedHeader == null)
        return false;

    return Math.Max(poll.PollStartBlockData.Height + network.ConsensusOptions.PollExpiryBlocks, network.ConsensusOptions.Release1100ActivationHeight) <= chainedHeader.Height;
}
```

**Comment by quantumagi** [comment]
LGTM. Perhaps just the one suggested change and CI must pass.

File: src/Stratis.Bitcoin.Features.PoA/Voting/PollsRepository.cs (hunk 1 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.PoA/Voting/PollsRepository.cs]: @quantumagi fixed, please re-check.

**Review by quantumagi — APPROVED** [review]
LGTM

**Pull request merged**